components/org.wso2.carbon.identity.conditional.auth.config.entgra/src/main/java/org/wso2/carbon/identity/conditional/auth/config/entgra/EntgraConfigImpl.java

```diff
@@ -0,0 +1,138 @@
+
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.conditional.auth.config.entgra;
+
+import org.wso2.carbon.identity.governance.IdentityGovernanceException;
+import org.wso2.carbon.identity.governance.common.IdentityConnectorConfig;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+public class EntgraConfigImpl implements IdentityConnectorConfig {
+
+    public static final String ENABLE = "adaptive_authentication.entgra.enable";
+    public static final String TOKEN_URL = "adaptive_authentication.entgra.token_url";
+    public static final String DEVICE_INFO_URL = "adaptive_authentication.entgra.device_info_url";
+    public static final String CLIENT_KEY = "adaptive_authentication.entgra.client_key";
+    public static final String CLIENT_SECRET = "adaptive_authentication.entgra.client_secret";
+
+    public static final String DEFAULT_ENABLE = "true";
+    public static final String DEFAULT_TOKEN_URL = "https://500.mgt.entgra.net/oauth/token";
+    public static final String DEFAULT_DEVICE_INFO_URL = "https://500.gw.entgra.net/api/device-mgt/v1.0/devices/1.0.0";
+    public static final String DEFAULT_CLIENT_KEY = "change-me";
+    public static final String DEFAULT_CLIENT_SECRET = "change-me";
+
+    @Override
+    public String getName() {
+
+        return "entgra-config";
+    }
+
+    @Override
+    public String getFriendlyName() {
+
+        return "Entgra Configuration";
+    }
+
+    @Override
+    public String getCategory() {
+
+        return "Other Settings";
+    }
+
+    @Override
+    public String getSubCategory() {
+
+        return "DEFAULT";
+    }
+
+    @Override
+    public int getOrder() {
+
+        return 11;
+    }
+
+    @Override
+    public Map<String, String> getPropertyNameMapping() {
+
+        Map<String, String> mapping = new HashMap<>();
+
+        mapping.put(ENABLE, "Enable Entgra");
+        mapping.put(TOKEN_URL, "Token URL");
+        mapping.put(DEVICE_INFO_URL, "Device Information URL");
+        mapping.put(CLIENT_KEY, "Client Key");
+        mapping.put(CLIENT_SECRET, "Client Secret");
```

Comment:
Why is a new map getting generated every time the getter is called?

Reply:
Will add it